substrate/primitives/metadata-ir/src/types.rs

```diff
@@ -525,3 +494,32 @@ impl IntoPortable for DeprecationStatus {
 		}
 	}
 }
+/// Deprecation info for an enums/errors/calls.
+/// Denotes full/partial deprecation of the typ
+#[derive(Clone, PartialEq, Eq, Encode, Debug)]
+pub enum DeprecationInfo<T: Form = MetaForm> {
```

Comment:
nit: I would name this `DeprecationInfoIR`, since it will make everything clearer from the `frame-metadata` crate perspective

Reply:
done for both